core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -515,7 +520,8 @@ class ReplicaManager(val config: KafkaConfig,
             case Some(p) =>
               if (p eq ReplicaManager.OfflinePartition)
                 throw new KafkaStorageException("Partition %s is in an offline log directory on broker %d".format(topicPartition, localBrokerId))
-              p
+              val (partition, _) = getPartitionAndLeaderReplicaIfLocal(topicPartition)
```

Comment:
You can remove a bunch of the code in this method now, `getPartition`, `partition.leaderReplicaIfLocal`, the `OfflinePartition` check. All of that is done in `getPartitionAndLeaderReplicaIfLocal`.